libbeat/reader/syslog/message.go

```diff
@@ -0,0 +1,220 @@
+	}
+}
+
+// setPriority sets the priority for this message. The facility and severity are
+// derived from the priority and associated values are set.
+func (m *message) setPriority(v string) {
+	m.priority = stringToInt(v)
+	m.facility = m.priority >> facilityShift
+	m.severity = m.priority & severityMask
+}
+
+// setHostname sets the hostname for this message. If the value is the "nil value" (-), the hostname will NOT be set.
+func (m *message) setHostname(v string) {
+	if v != "-" {
+		m.hostname = v
+	}
+}
+
+// setMsg sets the msg for this message. If the message includes a UTF-8 byte order mark (BOM),
+// then it will be removed.
+func (m *message) setMsg(v string) {
+	m.msg = strings.TrimPrefix(v, utf8BOM)
+}
+
+// setTag sets the process for this message.
+func (m *message) setTag(v string) {
+	m.process = v
+}
+
+// setAppName sets the process for this message. If the value is the "nil value" (-), the process will NOT be set.
+func (m *message) setAppName(v string) {
+	if v != "-" {
+		m.process = v
+	}
+}
+
+// setContent sets the pid for this message.
+func (m *message) setContent(v string) {
+	m.pid = v
+}
+
+// setProcID sets the pid for this message. If the value is the "nil value" (-), the pid will NOT be set.
+func (m *message) setProcID(v string) {
+	if v != "-" {
+		m.pid = v
+	}
+}
+
+// setMsgID sets the msgID for this message. If the value is the "nil value" (-), the msgID will NOT be set.
+func (m *message) setMsgID(v string) {
+	if v != "-" {
+		m.msgID = v
+	}
+}
+
+// setVersion sets the version for this message.
+func (m *message) setVersion(v string) {
+	n := stringToInt(v)
+	m.version = n
+}
+
+// setDataValue sets a structured data value. 'id' must already exist in the
+// structured data map, otherwise the value will not be set.
+func (m *message) setDataValue(id, key, value string) {
+	if _, okID := m.structuredData[id]; okID {
+		m.structuredData[id][key] = value
+	}
+}
+
+// fields produces fields from the message.
+func (m message) fields(original string) common.MapStr {
+	f := common.MapStr{}
+
+	// Syslog fields.
+	f.Put("log.syslog.priority", m.priority)
+	f.Put("log.syslog.facility.code", m.facility)
+	f.Put("log.syslog.severity.code", m.severity)
+	if v, ok := mapIndexToString(m.severity, severityLabels); ok {
+		f.Put("log.syslog.severity.name", v)
+	}
+	if v, ok := mapIndexToString(m.facility, facilityLabels); ok {
+		f.Put("log.syslog.facility.name", v)
+	}
+	if m.process != "" {
+		f.Put("log.syslog.appname", m.process)
+		if m.pid != "" {
+			f.Put("log.syslog.procid", m.pid)
+		}
+	}
+	if m.hostname != "" {
+		f.Put("log.syslog.hostname", m.hostname)
+	}
+	if m.msgID != "" {
+		f.Put("log.syslog.msgid", m.msgID)
+	}
+	if m.version != 0 {
+		f.Put("log.syslog.version", m.version)
+	}
+	if len(m.structuredData) > 0 {
+		f.Put("log.syslog.structured_data", m.structuredData)
+	}
+
+	// Event fields.
+	f.Put("event.severity", m.severity)
+	if original != "" {
+		f.Put("event.original", original)
```

Comment:
If `event.original` already exists then it should probably leave it as is.

Reply:
Maybe it shouldn't even be responsible for setting this. Users can do this on their own via copy or rename. This is how decode_cef expects it to be done. See the example at https://www.elastic.co/guide/en/beats/filebeat/8.1/processor-decode-cef.html#processor-decode-cef.

So an example of parsing a syslog message containing CEF data would be:


```
processors:
  - rename:
      fields:
        - {from: "message", to: "event.original"}
  - syslog:
      field: event.original
  - decode_cef:
      field: message
```

Reply:
I agree. It's easy enough for a user to just copy the message field if they want, and gives them control over what `event.original` should represent.